app/models/authentication.rb

```diff
@@ -68,6 +68,26 @@ def self.db_name
     "auth_key_pair_cloud"
   end
 
+  # Builds a case statement that case be used in a sql ORDER BY.
+  #
+  # Generated SQL looks like:
+  #
+  #   CASE
+  #   WHEN LOWER(status) = ''      THEN -1
+  #   WHEN LOWER(status) = 'valid' THEN 0
+  #   ...
+  #   ELSE -1
+  #
+  def self.status_severity_arel
+    arel_case = Arel::Nodes::Case.new
+
+    STATUS_SEVERITY.each do |value, sort_weight|
+      arel_case.when(arel_table[:status].lower.eq(value)).then(sort_weight)
+    end
+
+    arel_case.else(-1)
```

Comment:
This entire method is essentially constant right?  I mean it's built entirely on constant information.  Does it make sense to also make it a constant and/or memoize?  Or is rebuilding it just fine?

Reply:
I was actually considering it as a constant, but it was originally going to look really gross as such, so I decided to make it a method.  I was also originally `COALESCE`ing the values since I thought it was necessary, but it turns out it is not (the `else` and empty string cases cover that just fine).

I might experiment real quick rebuilding it, along with some specs that I forgot I was going to add, so I will see what it ends up like.

It shouldn't really be a big deal to rebuild it every time (performance wise), since latency from the DB is probably much a big hit in just about every scenario.  That said, it won't hurt turning it into a constant either.  The only reservation that might cause this to be an issue when used as a constant is if this was ever used when the table is aliased (for self-referencing or otherwise).  Example:

```SQL
SELECT *
FROM authentications AS auth
JOIN authentications AS other_auth ON auth.resource_id = other_auth.resrouce_id ...
```

but I haven't attempted to solve that case with this method easier, so probably isn't worth mentioning.